src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -2816,6 +2816,8 @@ address MacroAssembler::trampoline_call(Address entry, CodeBuffer* cbuf) {
          entry.rspec().type() == relocInfo::static_call_type ||
          entry.rspec().type() == relocInfo::virtual_call_type, "wrong reloc type");
 
+  address target = entry.target();
```

Comment:
I see we are call far_branches() and entry.target() more than once here in this function, which could be simplified.
I think we can set target according to result of far_branches() and plant a single jal(target) instead.

Reply:
> We call far_branches() and entry.target() more than once here in this function, which could be simplified. I think we can set target according to result of far_branches() and plant a single jal(target) instead.

Done. Thanks for reviewing!